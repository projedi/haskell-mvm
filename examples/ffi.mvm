dlopen("libm.so.6");

foreign float sin(float x);

print(sin(3));

float sqrt(float x);
foreign float sqrt(float x);

print(sqrt(3));

{
  float sin(float x) {
    print("Fake");
    return 0;
  }
  print(sin(3));
}

float cos(float x) {
  print("Fake");
  return 0;
}

{
  foreign float cos(float x);
  print(cos(3));
}
