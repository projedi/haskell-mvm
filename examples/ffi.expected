0.141
1.732
Fake
0
-0.99
